src/AdyenCheckoutContext.tsx

```diff
@@ -219,66 +213,74 @@ const AdyenCheckout: React.FC<AdyenCheckoutProps> = ({
         );
       }
 
-      if (isAddressLooker(nativeComponent)) {
+      const onUpdateAddressCallback = configuration.card?.onUpdateAddress;
+      const onConfirmAddressCallback = configuration.card?.onConfirmAddress;
+      if (
+        onUpdateAddressCallback &&
+        onConfirmAddressCallback &&
+        nativeComponent.events.includes(Event.onAddressUpdate) &&
+        nativeComponent.events.includes(Event.onAddressConfirm)
+      ) {
         const nativeModule = nativeComponent as unknown as AddressLookup;
         subscriptions.current.push(
           eventEmitter.addListener(
             Event.onAddressUpdate,
             async (prompt: string) => {
-              configuration.card?.onUpdateAddress?.(prompt, nativeModule);
+              onUpdateAddressCallback(prompt, nativeModule);
             }
           ),
           eventEmitter.addListener(
             Event.onAddressConfirm,
             (address: AddressLookupItem) => {
-              configuration.card?.onConfirmAddress?.(address, nativeModule);
+              onConfirmAddressCallback(address, nativeModule);
             }
           )
         );
       }
 
-      if (isPartialPaymentsComponent(nativeComponent)) {
+      const onBalanceCheckCallback =
+        configuration.partialPayment?.onBalanceCheck;
+      const onOrderRequestCallback =
+        configuration.partialPayment?.onOrderRequest;
+      const onOrderCancelCallback = configuration.partialPayment?.onOrderCancel;
+      if (
+        onBalanceCheckCallback &&
+        onOrderRequestCallback &&
+        onOrderCancelCallback &&
+        nativeComponent.events.includes(Event.onCheckBalance) &&
+        nativeComponent.events.includes(Event.onRequestOrder) &&
+        nativeComponent.events.includes(Event.onCancelOrder)
+      ) {
+        let component = nativeComponent as unknown as PartialPaymentComponent;
```

Comment:
Seems like component can be `const` if it is not reassigned

Reply:
Fixed in [9333374](https://github.com/Adyen/adyen-react-native/pull/599/commits/9333374bfb1fb058e7115d9134c73000082fddc9)